### ISSUE TITLE
Email sent as html format

### ISSUE THREAD

**Issue by paritoshbotre** [the-refinery/sparkpost_rails#14]
This is my config/environments/development.rb

    config.action_mailer.delivery_method = :sparkpost
    config.action_mailer.smtp_settings = {
        :address              => 'smtp.sparkpostmail.com',
        :port                    => 587,
        :domain               => 'gmail.com', #you can also use google.com
        :enable_starttls_auto => true,
        :encryption          => 'STARTTLS'
      }

I put my ```api_key``` in ```config/initializer/sparkpost_rails.rb```

i getting my email but my it uses html format

![issue_sparkpost](https://cloud.githubusercontent.com/assets/8109523/14638003/b1ccddfe-0652-11e6-969e-03e25b9c021a.png)

**Comment by dgoerlich** [comment]
A couple quick follow-up questions - are you wanting to leverage the SparkPost API to send the email or the SMTP server?  If it's the former, you probably should remove the smtp_settings block entirely.  I'm not sure what ActionMailer will do with both that and the delivery_method defined.

Also, is your issue that you are receiving HTML email, or that your HTML is being delivered in the plain text portion of the message?

**Comment by hgani** [comment]
Hi @dgoerlich 

I think I am getting the same issue so let me explain on behalf of the original poster.

The issue is that the SparkPost sends HTML email in plain text format, so the recipient sees the HTML code instead of the nicely rendered page.

Upon checking the email header, I see: 

Content-Type: text/plain; charset="UTF-8"

Please suggest a solution.

Thanks

**Comment by hgani** [comment]
I think I found the cause.

If we only provide a single .html ActionMailer view, this gem will assume that it's plain text. So the solution is to also provide a .text file (making it a multipart).

P.S. For those who don't want to manually write a .text file, you can use a gem that auto-generates it from the .html counterpart (e.g. https://github.com/lucaspiller/plain-david)

**Comment by dgoerlich** [comment]
Ah yeah - this makes sense.  It's the gap between SparkPost's abilities and how ActionMailer works.  Looking here in the code:  https://github.com/the-refinery/sparkpost_rails/blob/master/lib/sparkpost_rails/delivery_method.rb#L143  We pass the HTML part (mail.html_part.body) if it exists in the mail object, but only if the mail object reports as "multi-part".  If it doesn't the content (from the template) is in a different attribute (mail.body) and it's not really possible to tell if that's HTML content or plain text.  From what research I did, I found it's "best practice" to send plaintext content as well when sending HTML email.

That said, we could add an option to support HTML-only emails, effectively adding a flag to indicate that the mail is not multi-part, and that the mail.body content should be sent as HTML.  I can get that added for those that might not want to use the gem that @hgani references.
